src/server.jl

```diff
@@ -953,7 +953,19 @@ function extract_cell_options(source::AbstractString; file::AbstractString, line
                   """
             error(msg)
         end
-        isa(options, Dict) || error("Cell attributes must be a dictionary.")
+        if !isa(options, Dict)
+            msg = """
+                  Invalid cell attributes type at $(file):$(line):
+
+                  ```{julia}
+                  $source
+                  ```
+
+                  Expected a dictionary, got $(typeof(options)). Check for
+                  syntax errors in the YAML block at the start of this cell.
+                  """
+            error(msg)
```

Comment:
This could be a more specific error type but it might not matter too much.

Reply:
Doesn't matter too much since these are mostly just surfaced in the terminal when used via `quarto` itself rather than programmatically. Maybe we'll revisit the decision in future, but for now it should be fine as is.